lib/rewards.js

```diff
@@ -23,7 +26,22 @@ export const runUpdateRewardsLoop = async ({ contracts, ethAddress, onMetrics })
   }
 }
 
-async function getScheduledRewardsWithFallback (contract, ethAddress) {
+async function getSparkRewardsScheduledRewardsWithFallback (ethAddress) {
+  try {
+    const res = await fetch(
+      `https://spark-rewards.fly.dev/scheduled-rewards/${ethAddress}`
+    )
+    const json = await res.json()
```

Comment:
How do you handle 4xx and 5xx responses? What if the server responds with 5xx and the response body provides error details? Should you check `res.ok`?

In other places, I had a great experience with the helper `assertOkResponse` that reads the error response body and includes it in `err.message`.

Reply:
> How do you handle 4xx and 5xx responses? What if the server responds with 5xx and the response body provides error details? Should you check res.ok?

I believe we can get to these what ifs when we need. For now, we know that it will throw when there's a bad response, and it will contain the error:

```
fetch('https://ohnothisdoesnotexist.beep').catch(console.error)
TypeError: fetch failed
    at node:internal/deps/undici/undici:13178:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  [cause]: Error: getaddrinfo ENOTFOUND ohnothisdoesnotexist.beep
      at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
      at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    errno: -3008,
    code: 'ENOTFOUND',
    syscall: 'getaddrinfo',
    hostname: 'ohnothisdoesnotexist.beep'
  }
}
```

`spark-rewards` does not return a JSON error in this route, so parsing the response is premature. We could include the response text in the error, but this route also doesn't do that - if an error happens, it returns `Internal Server Error`.

Reply:
FWIW:
> ```js
> fetch('https://ohnothisdoesnotexist.beep').catch(console.error)
> ```

That's a different error case. This is what I had in mind:

```
(await fetch('https://example.com')).json().catch(console.error)
SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:5329:19)
    at successSteps (node:internal/deps/undici/undici:5300:27)
    at fullyReadBody (node:internal/deps/undici/undici:1447:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async specConsumeBody (node:internal/deps/undici/undici:5309:7)
```